src/gui/modalMenu.cpp

```diff
@@ -298,26 +295,18 @@ bool GUIModalMenu::preprocessEvent(const SEvent &event)
 			if (event.TouchInput.Event == ETIE_LEFT_UP)
 				leave();
 			return ret;
-		}
-		case 2: {
-			if (event.TouchInput.Event != ETIE_PRESSED_DOWN)
+		} else if (event.TouchInput.ID == 1) {
+			if (event.TouchInput.Event != ETIE_LEFT_UP)
 				return true; // ignore
 			auto focused = Environment->getFocus();
 			if (!focused)
 				return true;
-			SEvent rclick_event{};
-			rclick_event.EventType = EET_MOUSE_INPUT_EVENT;
-			rclick_event.MouseInput.Event = EMIE_RMOUSE_PRESSED_DOWN;
-			rclick_event.MouseInput.ButtonStates = EMBSM_LEFT | EMBSM_RIGHT;
-			rclick_event.MouseInput.X = m_pointer.X;
-			rclick_event.MouseInput.Y = m_pointer.Y;
-			focused->OnEvent(rclick_event);
-			rclick_event.MouseInput.Event = EMIE_RMOUSE_LEFT_UP;
-			rclick_event.MouseInput.ButtonStates = EMBSM_LEFT;
-			focused->OnEvent(rclick_event);
+			m_second_touch = true;
```

Comment:
```suggestion
			// The second touch event is not converted into a mouse event, but
			// is propagated as is.
			m_second_touch = true;
```

I think this deserves a comment.